table/block_based/block_based_table_builder.cc

```diff
@@ -1138,7 +1141,7 @@ void BlockBasedTableBuilder::BGWorkWriteRawBlock() {
     }
 
     if (r->pc_rep->first_block) {
-      std::unique_lock<std::mutex> lock(r->pc_rep->first_block_mutex);
+      std::lock_guard<std::mutex> lock(r->pc_rep->first_block_mutex);
```

Comment:
What's the reason for the changing? If it is changed here, it feel natural to change the other one too.

Reply:
`unique_lock` provides an encapsulation of locking/unlocking a mutex, which is necessary for `first_block_cond.wait` call in the other place. `lock_guard` is sufficient here because we don't need the encapsulation here.